### PR TITLE
Feature/show survey site

### PR DESCRIPTION
**Pull request by Roland-Hufnagel**

### CODE REVIEW

File: components/QuestionCard.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import styled from "styled-components";
+
+export default function QuestionCard({ question }) {
+  return (
+    <StyledQuestionCard>
+      <legend>{question}</legend>
+      <input type="radio" id="yes" name={question} value="Yes" required/>
```
Comment (F-Kirchhoff) [inline components/QuestionCard.js:7]: you don't need that required attribute there

File: components/QuestionCard.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      <label htmlFor="yes">Yes</label>
+      <input type="radio" id="no" name={question} value="No" />
+      <label htmlFor="no">No</label>
+    </StyledQuestionCard>
+  );
+}
+
+const StyledQuestionCard = styled.fieldset`
+  border: 1px solid black;
+  margin: 1em auto;
```
Comment (F-Kirchhoff) [inline components/QuestionCard.js:17]: the outer spacing of an element should be handled by it's parent element

File: pages/index.js [hunk 3]
```diff
@@ -12,15 +13,10 @@ export default function Home() {
 
       <main>
         <h1>My App</h1>
-        <MyDiv>Welcome World</MyDiv>
+        
       </main>
     </div>
```
Comment (F-Kirchhoff) [inline pages/index.js:18]: Where do you use the QuestionCard?

File: pages/surveys/[id].js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  };
+}
+
+export default function Survey({ id, title, questions }) {
+  const router = useRouter();
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
```
Comment (F-Kirchhoff) [inline pages/surveys/[id].js:22]: right now nothing happens with the data, in the future make sure that everything you code serves a purpose

File: pages/surveys/[id].js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    router.push("./thankyou");
+  }
+
+  return (
+    <StyledForm onSubmit={handleSubmit}>
+      <h1>{title}</h1>
+      {questions.map((question, index) => {
+        return <QuestionCard key={index} question={question} />;
+      })}
```
Comment (F-Kirchhoff) [inline pages/surveys/[id].js:31]: you could use an implicit return here

Also, make sure that your questioncards are Li elements wrapped by an ul element

File: pages/surveys/[id].js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+const StyledForm = styled.form`
+  max-width: 600px;
+  margin: 0 auto;
+  & h1 {
+    text-align: center;
+  }
+  & button {
+    width: 100%;
+    font-size: 1.2em;
+  }
```
Comment (F-Kirchhoff) [inline pages/surveys/[id].js:46]: nice!

File: pages/surveys/thankyou/index.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import styled from "styled-components";
+
+export default function Thankyou() {
+  return (
+    <StyledDiv>
+      <div>Thank you for your feedback</div>
```
Comment (F-Kirchhoff) [inline pages/surveys/thankyou/index.js:6]: Please use a p tag here

File: pages/surveys/thankyou/index.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    </StyledDiv>
+  );
+}
+
+const StyledDiv = styled.div`
```
Comment (F-Kirchhoff) [inline pages/surveys/thankyou/index.js:11]: Can you find a better element than 'div' here?

File: services/surveyService.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        title: "cgn-web-22-4 weekly feedback week 3",
+        questions: ["Are you okay?", "Do you like Music?", "Do you like Elvis Presley?", "Do you like Ray Charles?"]
+    },
+]
+
+
+export async function getSurveyById(id){
+    const survey = surveys.find((survey)=>survey.id == id);
+    return survey;
+}
```
Comment (F-Kirchhoff) [inline services/surveyService.js:23]: Why is this an async function?

File: pages/surveys/[id].js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+import { useRouter } from "next/router";
+
+export async function getServerSideProps(context) {
+  const { id } = context.params;
+  const survey = await getSurveyById(id);
+  return {
+    props: {
+      ...survey,
+    },
+  };
```
Comment (F-Kirchhoff) [inline pages/surveys/[id].js:13]: This doesn't need to be async since there are no Promises that need to resolve in your code.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Roland, thats already a lot of code, and most of it quite good. I still found some stuff to improve ;) Make sure for the next PRs that you don't code anything that doesn't fit a purpose in your App, you can always add it with the next PR

File: components/QuestionCard.js (hunk 1 above)
Comment (Roland-Hufnagel) [reply components/QuestionCard.js:7]: yes

File: components/QuestionCard.js (hunk 2 above)
Comment (Roland-Hufnagel) [reply components/QuestionCard.js:17]: yes, but spacing between siblings have to be declared here.

File: services/surveyService.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        id: "abc3",
+        title: "cgn-web-22-4 weekly feedback week 3",
+        questions: ["Are you okay?", "Do you like Music?", "Do you like Elvis Presley?", "Do you like Ray Charles?"]
+    },
+]
+
+
+export  function getSurveyById(id){
+    const survey = surveys.find((survey)=>survey.id == id);
```
Comment (CathrinTruchan) [inline services/surveyService.js:21]: You could use the strict equality operator here (===)

**Review by CathrinTruchan — APPROVED** [review]

**Review by CathrinTruchan — APPROVED** [review]

**Pull request merged**